tests/util_tests.py

```diff
@@ -33,3 +35,71 @@ def test_iso_gregorian(self):
 
         with self.assertRaises(ValueError):
             util.iso_to_gregorian(2013, 8, 0)
+
+    def test_windows_datetime_from_timestamp(self):
+        timestamp = 1572204340.6460679
+        result = util.windows_datetime_from_timestamp(timestamp)
+        expected = datetime.fromtimestamp(timestamp).replace(tzinfo=tz.tzlocal())
+        self.assertEqual(result, expected)
+
+    def test_windows_datetime_from_timestamp_utc(self):
+        timestamp = 1572204340.6460679
+        result = util.windows_datetime_from_timestamp(timestamp, tz.tzutc())
+        expected = datetime.utcfromtimestamp(timestamp).replace(tzinfo=tz.tzutc())
+        self.assertEqual(result, expected)
+
+    def test_safe_utcfromtimestamp(self):
+        timestamp = 1572204340.6460679
+        result = util.safe_utcfromtimestamp(timestamp).replace(tzinfo=tz.tzutc())
+        expected = datetime.utcfromtimestamp(timestamp).replace(tzinfo=tz.tzutc())
+        self.assertEqual(result, expected)
+
+    def test_safe_fromtimestamp_default_tz(self):
+        timestamp = 1572204340.6460679
+        result = util.safe_fromtimestamp(timestamp).replace(tzinfo=tz.tzlocal())
+        expected = datetime.fromtimestamp(timestamp).replace(tzinfo=tz.tzlocal())
+        self.assertEqual(result, expected)
+
+    def test_safe_fromtimestamp_paris_tz(self):
+        timestamp = 1572204340.6460679
+        result = util.safe_fromtimestamp(timestamp, tz.gettz("Europe/Paris"))
+        expected = datetime.fromtimestamp(timestamp, tz.gettz("Europe/Paris"))
+        self.assertEqual(result, expected)
+
+    def test_safe_utcfromtimestamp_negative(self):
+        timestamp = -1572204340.6460679
+        result = util.safe_utcfromtimestamp(timestamp).replace(tzinfo=tz.tzutc())
+        expected = datetime(1920, 3, 7, 4, 34, 19, 353932, tzinfo=tz.tzutc())
+        self.assertEqual(result, expected)
+
+    def test_safe_fromtimestamp_negative(self):
+        timestamp = -1572204340.6460679
+        result = util.safe_fromtimestamp(timestamp, tz.gettz("Europe/Paris"))
+        expected = datetime(
+            1920, 3, 7, 5, 34, 19, 353932, tzinfo=tz.gettz("Europe/Paris")
+        )
+        self.assertEqual(result, expected)
+
+    def test_safe_utcfromtimestamp_negative_nt(self):
+        timestamp = -1572204340.6460679
+        result = util.safe_utcfromtimestamp(timestamp, os_name="nt").replace(
```

Comment:
Instead of passing in `os_name` into this function, let's instead use the Python mock library (already in `requirements.txt`) to patch `os.name` to "nt". This way, we can remove the `os_name` argument since it probably won't be used outside of unit tests.

Or, we can only run this test when `os.name == "nt"`.

Reply:
Ok, will try that